### DISCUSSION TITLE
Support juicefs sync with src http and dst https

### DISCUSSION THREAD

**Discussion by deepmap-haonan** [juicedata/juicefs#4905] [General]
**What would you like to be added**:
juicefs sync with source http and destination https. Currently, sync can be used with both http with --no-https.

**Why is this needed**:
We have an environment mixed the usage of https and http. This caused two object storages system use http only or https only.

For example, for source of sync we are using endpoint `http://s3.qingstor.endpoint` and for the destination of sync we are using `https://s3.swiftstack.endpoint`. If we sync with `no-https`, sync will fail due to cannot connect with swiftstack. And vise versa without `no-https`.

**Comment by davies** [comment]
We may need two parameters to disable HTTPS for them separately, but I think it's too much for this rarely use-case, I'd suggest that you patch JuiceFS to make it work or add HTTPS for the source or allow HTTP for the destination as a workround.
